graph/path/yen_ksp.go

```diff
@@ -111,7 +111,16 @@ func YenKShortestPaths(g graph.Graph, k int, cost float64, s, t graph.Node) [][]
 		if len(best.path) <= 1 || best.weight > cost {
 			break
 		}
+		if neededPaths := k - i; k >= 0 && len(pot) >= neededPaths && best.weight == pot[neededPaths-1].weight {
```

Comment:
This is the second optimisation? These should be in separate commits.

Reply:
Yes. Done.